src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1302,7 +1301,7 @@ void VideoDecoder::convertAVFrameToFrameOutputOnCPU(
 }
 
 int VideoDecoder::convertAVFrameToTensorUsingSwsScale(
-    const AVFrame* avFrame,
+    UniqueAVFrame& avFrame,
```

Comment:
Unless I'm misunderstanding, pretending the `AVFrame*` was `const` was a big lie, because of the call to `sws_scale(..., avFrame->data, ...)` below. So I made the ref non-const.

Reply:
I think it still makes sense for it to be const. We have a const reference to a unique pointer, which means that we can't change the contents of the unique pointer. That is, we can't call `release()` or `reset()` on the unique pointer. But we can make changes to the object the unique pointer points to.

This is similar to the difference between a "const pointer" and "pointer to const." A const pointer is a pointer that cannot change what it is pointing to. A pointer to const is a pointer to an object that cannot change - but you can change what the pointer points to.

Basically, anytime we don't want _ownership_ of the unique pointer to change, we should use const.